book/core_programming.md

```diff
@@ -2,153 +2,310 @@
 
-Before reading this chapter, you would benefit from having an understanding of core programming concepts.
-This includes assigning and using variables, and using functions and classes.
+However, they are tailored to a more analytical workflow and would benefit analysts with understanding of core programming concepts such as variables, functions and classes. You can find more information and helpful resources in the [resources](resources) section.
 
+## Motivation
 
-```{admonition} Pre-requisites
-:class: admonition-learning
+```{epigraph}
+Code is read more often than it is written.
 
-You should look to these sources for introductory learning, before reading this guidance:
-* Introduction to [Python](https://learninghub.ons.gov.uk/enrol/index.php?id=536) and [R](https://learninghub.ons.gov.uk/enrol/index.php?id=538) (government analysts only)
-* Aggregation, functions and control flow in [Python](https://learninghub.ons.gov.uk/enrol/index.php?id=525) and [R](https://learninghub.ons.gov.uk/enrol/index.php?id=527) (government analysts only)
-* Official Python [Getting Started Guide](https://www.python.org/about/gettingstarted/)
-* [Learn Python](https://www.learnpython.org/) supported by DataCamp
-* [R Basics](https://www.udemy.com/course/r-basics/?LSNPUBID=JVFxdTr9V80&ranEAID=JVFxdTr9V80&ranMID=39197&ranSiteID=JVFxdTr9V80-mzkL.lpta.ugiFro.8Oc_Q&utm_medium=udemyads&utm_source=aff-campaign) on Udemy
-* [edx](https://www.edx.org/) and [freeCodeCamp](https://www.freecodecamp.org/) hosts many free online coding courses
-* [DataCamp](https://www.datacamp.com/), [Codecademy](https://www.codecademy.com/) and [Coursera](https://www.coursera.org/) host a range of introductory courses, but more advanced courses are often not free.
+-- Guido van Rossum (creator of Python)
+```
 
-Please note that non-government analysis learning may not follow good practices. However, exposure to a range of applied examples will still benefit your learning.
-You should compare and contrast your learning to the good practices outlined in this section.
+When writing non-trivial code it is wise to assume that at some point someone else will need to understand, use and adapt your code. It might be yourself in six months time - after you've forgotten how the code works, or its design intent. Therefore every time you write such code, it is incredibly important to empathise with these potential users and produce code that is tidy, understandable and does not add unnecessary complexity.
+
+Common barriers to writing readable codebases include documentation that is hard to understand or absent, walls of code with repeating functionality that is hard to absorb in 'chunks' or overcomplicated solutions that solve the problem in ways that could be simplified. Avoiding these issues is essential to make sure that your analysis is reproducible, auditable and assured. Therefore it is our professional responsibility to avoid putting such barriers in place whenever possible.
```

Comment:
Are these barriers to writing readable codebases, or are they barriers to reading codebases? I don't think we should talk specifically about professional responsibility here - just a general sense of responsibility.

Reply:
Well, I think making it about - this is part of your job not just part of your character is probably a stronger message to leave people with.

Reply:
Sort of hinting that writing good code is part of the job description.